### DISCUSSION TITLE
[Q] Make OPAM recognize this existing switch?

### DISCUSSION THREAD

**Discussion by anzhi0708** [ocaml/ocaml#12110] [Q&A]
After compiling OCaml 5.1 from source, OPAM still recognizes **only** previously installed switch

```
~ $ opam switch list
#  switch  compiler
          description
→  4.14.1  ocaml-base-compiler.4.14.1
          4.14.1
```

I compiled & installed OCaml 5 to `bin/` tho

```
~ $ ocaml --version
The OCaml toplevel, version 4.14.1
~ $ $PREFIX/bin/ocaml --version
The OCaml toplevel, version 5.1.0+dev1-2022-06-09
```
I want version 5.1 listed in `opam switch list` too.

These two versions have different installation directories
```
~ $ which ocaml
/data/data/com.termux/files/home/.opam/4.14.1/bin/ocaml
~ $ which $PREFIX/bin/ocaml
/data/data/com.termux/files/usr/bin/ocaml
```

Send help :3

**Comment by nojb** [comment]
Did you try the instructions in https://github.com/ocaml/ocaml/blob/trunk/HACKING.adoc#testing-with-opam?

**Reply by anzhi0708** [reply]
Thanks, I'll look into it! :3

**Comment by abbysmal** [comment]
It may also be simpler to use the `5.1.0+trunk` switch available in OPAM, if you just want to check the latest advancements and are not actively developing on the codebase.

**Reply by anzhi0708** [reply]
> It may also be simpler to use the `5.1.0+trunk` switch available in OPAM, if you just want to check the latest advancements and are not actively developing on the codebase.

Well, I could not, because those multicore version compilers listed in `opam switch list-available` currently are not able to fit Android platform very well; something in `configure.ac` caused it. I had to manually compile OCaml 5+ from source code, adding some specific flags here and there (thanks to OCaml dev team for helping). I'm installing OCaml 5 on my Android mobile phone :-(
